scripts/extract_tybalt_weights.py

```diff
@@ -0,0 +1,182 @@
+weight_layer_df.head(2)
+
+
+# In[8]:
+
+# Write the genes to file
+weight_file = os.path.join('results', 'tybalt_gene_weights.tsv')
+weight_layer_df.to_csv(weight_file, sep='\t')
+
+
+# ## Extracting example patterns learned by Tybalt
+# 
+# Focusing on two examples: Sex-specific and tissue-specific activation
+# 
+# ### Sex specific activation by node 82
+
+# In[9]:
+
+# We previously identified node 82 as robustly separating sex in the data set:
+# Visualize the distribution of gene weights here
+sex_node = weight_layer_df.loc[82, :]
+sex_node_plot = weight_layer_df.loc[[82, 85], :].T
+sex_node_plot.columns = ['encoding 82', 'encoding 85']
+
+sex_node_plot = (
+    sex_node_plot.reindex(sex_node_plot['encoding 82'].abs()
+                          .sort_values(ascending=False).index)
+    )
+
+g = sns.jointplot(x='encoding 82', y='encoding 85',
+                  data=sex_node_plot, color='black',
+                  edgecolor="w", stat_func=None);
+
+# Save Figure
+sex_node_plot_file = os.path.join('figures', 'sex_node_gene_scatter.pdf')
+g.savefig(sex_node_plot_file)
+
+
+# In[10]:
+
+# There are 17 genes with high activation in node 82
+# All genes are located on sex chromosomes
+sex_node_plot.head(17)
+
+
+# By measuring expression of only *17 genes*, we can reliably predict the sex of the cancer patient. These are genes most expressed by sex chromosomes including x inactivating genes _XIST_ and _TSIX_.
+
+# ### Node separating melanoma samples
+
+# In[11]:
+
+# We previously observed metastasis samples being robustly separated by two features
+# Visualize the feature scores here
+met_node_plot = weight_layer_df.loc[[53, 66], :].T
+met_node_plot.columns = ['encoding 53', 'encoding 66']
+
+met_node_plot = (met_node_plot.reindex(met_node_plot['encoding 53'].abs()
+                                       .sort_values(ascending=False).index)
+                 )
+g = sns.jointplot(x='encoding 53', y='encoding 66',
+                  data=met_node_plot, color='black',
+                  edgecolor="w", stat_func=None);
+
+# Save outputs
+met_node_plot_file = os.path.join('figures', 'skcm_metastasis_node_gene_scatter.pdf')
+g.savefig(met_node_plot_file)
+
+
+# #### Output high weight genes for two specific SKCM encodings
+# 
+# Because the genes involved in these nodes are not as cut and dry as the sex specific nodes, output the high weight genes explaining each tail of node 53 and 66. These will be processed though a pathway analysis downstream.
+
+# In[12]:
+
+def output_high_weight_genes(weight_df, encoding, filename, thresh=2):
+    """
+    Function to process and output high weight genes given specific feature encodings
+    """
+    
+    # Sort initial encoding by absolute activation
+    encoding_df = (
+        weight_df
+        .reindex(weight_df[encoding].abs()
+                 .sort_values(ascending=False).index)[encoding]
+    )
+    
+    hw_pos_df = pd.DataFrame(encoding_df[encoding_df > encoding_df.std() * thresh])
```

Comment:
Isn't this already a DataFrame? You don't need to recast it. Same for below

Reply:
it is actually a `Series`, not a `DataFrame` - I am recasting it because I am interested in using the DataFrame specific [`assign()`](https://pandas.pydata.org/pandas-docs/stable/generated/pandas.DataFrame.assign.html) method to add a column to each high weight gene object.